subtrack.MAUI/Pages/SubscriptionDetails.razor

```diff
@@ -0,0 +1,97 @@
+﻿@page "/details/{Id:int}"
+
+@inject ISubscriptionService subscriptionService
+@inject NavigationManager navigationManager
+@inject ISubscriptionsCalculator subscriptionsCalculator
+
+@if(_subscription is null)
+{
+    <div>Subscription details not found</div>
+} else
+{
+    <div class="container">
+        <div class="row border-bottom py-2">
+            <div class="col-10">
+                <h2>@_subscription.Name</h2>
+            </div>
+            <div class="col-2">
+                <button aria-label="Edit" type="button" class="btn"><i class="bi bi-pencil-square"></i></button>
+            </div>
+        </div>
+        <div class="row border-bottom py-2">
+            <div class="col-6 col-md-4">
+                <h5>@($"{_subscription.Cost:C}")</h5>
+            </div>
+            @if (_subscription.IsAutoPaid)
+            {
+                <div class="col-6 col-md-8">
+                    <i class="fa fa-repeat" aria-hidden="true"></i>
+                </div>
+            }
+        </div>
+        <div class="row border-bottom py-2">
+            <div class="col-12">
+                @{
+                    var nextPaymentDate = _subscription.LastPayment.AddDays(subscriptionsCalculator.GetDueDays(_subscription));
```

Comment:
prefer if this logic is done inside the OnInit method and might be nice to have it inside the SubscriptionCalculator in order to make it testable. GetNextPaymentDate()

Reply:
Done